Content.Server/GameObjects/Components/Items/FloorTileItemComponent.cs

```diff
@@ -0,0 +1,92 @@
+﻿using Content.Server.GameObjects.EntitySystems;
+using Content.Shared.Maps;
+using Robust.Server.GameObjects.EntitySystems;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Interfaces.Map;
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+using Robust.Shared.Map;
+using Robust.Shared.Serialization;
+using Robust.Shared.Utility;
+
+namespace Content.Server.GameObjects.Components.Items
+{
+    [RegisterComponent]
+    public class FloorTileItemComponent : Component, IAfterAttack, IExamine
+    {
+#pragma warning disable 649
+        [Dependency] private readonly ITileDefinitionManager _tileDefinitionManager;
+        [Dependency] private readonly IEntitySystemManager _entitySystemManager;
+        [Dependency] private readonly IMapManager _mapManager;
+#pragma warning restore 649
+
+        public override string Name => "FloorTile";
+        public int CurrentStackCount = 1;
```

Comment:
There is already a `StackComponent` that you can use to handle the whole stacking business.

Reply:
🤦‍♀